core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -704,9 +708,15 @@ public boolean isCoordinator() {
          return cache;
       } catch (CacheException e) {
          cacheFuture.completeExceptionally(e);
+         //remove the CompletableFuture in case of failure (example, X-Site check that happens during startup)
+         //to give a change to create the cache properly
+         caches.remove(cacheName, cacheFuture);
```

Comment:
-1 from me, if a cache failed to start then it should stay in the failed state until the user explicitly restarts it.

It seems wrong that `manager.getCache()` would get the same exception if called during the first `manager.getCache()`, but it would try to start the cache again if called 1ms later, possibly getting a different exception (e.g. because `ExceptionInInitializerError` becomes `NoClassDefFoundError`).

Reply:
I don't see how a user can restart a cache with an invalid configuration. We never remove caches from the `caches` Map